src/gui/Application.cpp

```diff
@@ -324,46 +327,40 @@ void Application::socketReadyRead()
         return;
     }
 
-    QStringList fileNames;
-    quint32 id;
+    SocketCmd id;
     in >> id;
 
-    // TODO: move constants to enum
     switch (id) {
-    case 1:
+    case SocketCmd::OpenFiles: {
+        QStringList fileNames;
         in >> fileNames;
         for (const QString& fileName : asConst(fileNames)) {
             const QFileInfo fInfo(fileName);
             if (fInfo.isFile() && fInfo.suffix().toLower() == "kdbx") {
                 emit openFile(fileName);
             }
         }
-
         break;
-    case 2:
+    }
+    case SocketCmd::LockAll:
         getMainWindow()->lockAllDatabases();
         break;
+    case SocketCmd::Unlock:
+        QString filename, password, keyfile;
+        in >> filename >> password >> keyfile;
+        emit openFile(filename, password, keyfile);
+        break;
     }
 
     socket->deleteLater();
 }
 
 bool Application::isAlreadyRunning() const
 {
-#ifdef QT_DEBUG
-    // In DEBUG mode we can run unlimited instances
-    return false;
-#endif
```

Comment:
This should not be removed

Reply:
My bad, that was not meant to be committed. Is there any better way to test single instance mode for debugging?

Reply:
Should I remove this again then?